modules/ballerina-core/src/main/java/org/ballerinalang/BLangProgramRunner.java

```diff
@@ -75,7 +75,7 @@ public static void runService(ProgramFile programFile) {
             }
 
             // Deploy service
-            ServerConnectorRegistry.getInstance().registerService(serviceInfo);
+            ServerConnectorRegistry.getInstance(programFile).registerService(serviceInfo);
```

Comment:
I think this is not good, get instance should not accept parameters. (BTW need to check whether we can remove this singleton as well)

Reply:
This is the same way the compiler has done the workaround to remove singleton. This cannot be removed since the data contains here is needed in multiple modules.
Maybe we can change the method name.

Reply:
I think you need to add another method. IMO getInstance implies getting the instance of the class itself, so it does not make sense to pass anything. 
so how about 
ServerConnectorRegistry.getInstance().getRegistry(programFile) 
something like that?

Reply:
As per the offline discussion with @lankavitharana we decided to remove this method and access server connector registry via programFile